### PR TITLE
Add VirtualMachineApi

### PR DESCRIPTION
**Pull request by jtjk**
Added VirtualMachineApi.

### CODE REVIEW

**Comment by jtjk** [comment]
Might be good idea to wait that other PRs are merged and I will rebase this against them and make proper setup and teardown to get rid off nic and storage configuration in test env.

**Comment by jtjk** [comment]
Rebased today to include new stuff from azurecompute-arm. It is now having proper setup and teardown to create resource group, storage and nic before starting actual tests. In the end it waits that VM is deleted to be able to tear down in base class.

**Comment by jtjk** [comment]
To run live tests: mvn -Dtest=VirtualMachineApiLiveTest -Dtest.azurecompute-arm.subscriptionid="" -Dtest.azurecompute-arm.identity="" -Dtest.azurecompute-arm.credential="" -Dtest.azurecompute-arm.endpoint="https://management.azure.com/" -Dtest.jclouds.oauth.resource="https://management.azure.com/" -Dtest.oauth.endpoint="https://login.microsoftonline.com/TENANT.onmicrosoft.com/oauth2/token" test

File: azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/VirtualMachineApi.java [hunk 1]
```diff
@@ -18,135 +18,111 @@
...
+   void start(@PathParam("name") String name);
 
    /**
-    * The Capture VM Image operation creates a copy of the operating system virtual hard disk (VHD) and all of the data
-    * VHDs that are associated with the Virtual Machine, saves the VHD copies in the same storage location as the original
-    * VHDs, and registers the copies as a VM Image in the image repository that is associated with the specified subscription.
+    * The stop Virtual Machine operation
     */
-   @Named("CaptureVMImage")
+   @Named("StopVirtualMachine")
```
Comment (ritazh) [inline azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/VirtualMachineApi.java:131]: Let's document and create a card for features we have no implemented in this api. For example: List all virtual machines in an availability set, list all available virtual machine sizes for resizing, save an image that is associated with a vm, stop an deallocate a vm.

File: azurecompute-arm/src/test/java/org/jclouds/azurecomputearm/features/VirtualMachineApiLiveTest.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+   @Test(dependsOnMethods = "testGet")
+   public void testList() {
+      List<VirtualMachine> list = api().list();
+      for (VirtualMachine machine : list) {
+         assertTrue(!machine.name().isEmpty());
+      }
+   }
+
+   @Test(dependsOnMethods = "testList")
```
Comment (ritazh) [inline azurecompute-arm/src/test/java/org/jclouds/azurecomputearm/features/VirtualMachineApiLiveTest.java]: Missing test cases for start, stop, restart

File: azurecompute-arm/src/test/java/org/jclouds/azurecomputearm/features/VirtualMachineApiMockTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+   private VirtualMachine getVM() {
+      VirtualMachineProperties properties = getProperties();
+      VirtualMachine machine = VirtualMachine.create("/subscriptions/626f67f6-8fd0-xxxx-yyyy-e3ce95f7dfec/" + "" +
+              "resourceGroups/groupname/providers/Microsoft.Compute/virtualMachines/windowsmachine", "windowsmachine",
+              "Microsoft.Compute/virtualMachines", "westus", null, properties);
+      return machine;
+   }
+
+   private List<VirtualMachine> getVMList() {
+      VirtualMachineProperties properties = getProperties();
```
Comment (ritazh) [inline azurecompute-arm/src/test/java/org/jclouds/azurecomputearm/features/VirtualMachineApiMockTest.java:147]: Missing test cases for start, stop, restart

**Comment by ritazh** [comment]
Tests run successfully. Missing few test cases.

**Comment by jtjk** [comment]
Added missing test cases

**Pull request merged**